File: articles/evolution-of-organizational-psychology/go/field_evolution_score.go
// The Evolution of Organizational Psychology
// Simple Go scoring demonstration.
//
// Responsible-use scope:
// Synthetic-data research, methods demonstration, organizational psychology
// education, and history-of-field analysis only. Not for employee, scholar,
// department, journal, or institution ranking.

package main

import "fmt"

type FieldEmphasis struct {
	SelectionTesting                  float64
	EfficiencyWorkDesign              float64
	MotivationAttitudes               float64
	GroupsTeams                       float64
	CultureLeadershipChange           float64
	WellbeingStress                   float64
	JusticeValidityEthics            float64
	DigitalAnalytics                  float64
	SystemsInstitutionalAccountability float64
}

func ResponsibleProgressScore(f FieldEmphasis) float64 {
	return 0.18*f.SelectionTesting +
		0.08*f.EfficiencyWorkDesign +
		0.12*f.MotivationAttitudes +
		0.12*f.GroupsTeams +
		0.14*f.CultureLeadershipChange +
		0.15*f.WellbeingStress +
		0.17*f.JusticeValidityEthics +
		0.08*f.DigitalAnalytics +
		0.06*f.SystemsInstitutionalAccountability
}

func main() {
	broad := FieldEmphasis{72, 66, 84, 86, 88, 86, 92, 84, 90}
	fmt.Printf("Synthetic responsible progress score: %.2f\n", ResponsibleProgressScore(broad))
	fmt.Println("Responsible-use reminder: educational field-evolution demo only.")
}

File: articles/evolution-of-organizational-psychology/cpp/field_evolution_score.cpp
/*
The Evolution of Organizational Psychology
Simple C++ scoring demonstration.

Responsible-use scope:
Synthetic-data research, methods demonstration, organizational psychology
education, and history-of-field analysis only. Not for employee or institution ranking.
*/

#include <iostream>

struct FieldEmphasis {
    double selection_testing;
    double efficiency_work_design;
    double motivation_attitudes;
    double groups_teams;
    double culture_leadership_change;
    double wellbeing_stress;
    double justice_validity_ethics;
    double digital_analytics;
    double systems_institutional_accountability;
};

double responsible_progress_score(const FieldEmphasis& f) {
    return
        0.18 * f.selection_testing +
        0.08 * f.efficiency_work_design +
        0.12 * f.motivation_attitudes +
        0.12 * f.groups_teams +
        0.14 * f.culture_leadership_change +
        0.15 * f.wellbeing_stress +
        0.17 * f.justice_validity_ethics +
        0.08 * f.digital_analytics +
        0.06 * f.systems_institutional_accountability;
}

int main() {
    FieldEmphasis broad{72, 66, 84, 86, 88, 86, 92, 84, 90};
    std::cout << "Synthetic responsible progress score: "
              << responsible_progress_score(broad) << "\n";
    std::cout << "Responsible-use reminder: educational field-evolution demo only.\n";
    return 0;
}
